core-web/libs/sdk/react/README.md

```diff
@@ -64,91 +114,132 @@ const MyPage = ({ page }) => {
 
 ## Hooks
 
-### `useDotcmsPageContext`
+### `useDotCMSShowWhen`
 
-A custom React hook that provides access to the `PageProviderContext`.
+A custom hook that provides the same functionality as the `DotCMSShow` component in a hook form. It uses the UVE_MODE from `@dotcms/uve` which can be one of: `EDIT`, `PREVIEW`, or `LIVE`.
 
-#### Returns
+#### Parameters
 
--   `PageProviderContext | null`: The context value or `null` if it's not available.
+| Parameter | Type | Required | Description |
+|-----------|------|----------|-------------|
+| `mode` | String | Yes | The UVE mode to check against (EDIT, PREVIEW, LIVE) |
 
 #### Usage
 
 ```javascript
-import { useDotcmsPageContext } from '@dotcms/react';
+import { useDotCMSShowWhen } from '@dotcms/react';
+import { UVE_MODE } from '@dotcms/uve';
 
 const MyComponent = () => {
-    const context = useDotcmsPageContext();
-    // Use the context
+    const isVisible = useDotCMSShowWhen(UVE_MODE.EDIT);
+    
+    return isVisible ? <div>Visible content</div> : null;
 };
 ```
 
-### `usePageEditor`
-
-A custom React hook that sets up the page editor for a dotCMS page.
+### `usePageAsset`
 
-#### Parameters
+A custom hook that handles the communication with the Universal View Editor (UVE) and updates your page content in real-time when changes are made in the editor.
 
--   **props**: `PageEditorOptions` - The options for the page editor. Includes a `reloadFunction` and a `pathname`.
+> **Note:** This hook will be built into the SDK in the stable version - the example below is a temporary workaround for the beta release.
+>
+> You need to save this hook in your project as a custom hook file.
 
-#### Returns
+#### Parameters
 
--   `React.RefObject<HTMLDivElement>[]`: A reference to the rows of the page.
+| Parameter | Type | Required | Description |
+|-----------|------|----------|-------------|
+| `currentPageAsset` | Object | Yes | The initial page asset from the server |
 
-#### Usage
+#### Implementation
 
 ```javascript
-import { usePageEditor } from '@dotcms/react';
+import { useEffect, useState } from 'react';
 
-const MyEditor = () => {
-    const rowsRef = usePageEditor({ pathname: '/my-page' });
-    // Use the rowsRef
-};
-```
-
-## Context Providers
+import { getUVEState, sendMessageToEditor, createUVESubscription} from '@dotcms/uve';
+import { DotCMSUVEAction, UVEEventType} from '@dotcms/uve/types';
 
-### `PageProvider`
+export const usePageAsset = (currentPageAsset) => {
+    const [pageAsset, setPageAsset] = useState(null);
+    useEffect(() => {
+        if (!getUVEState()) {
+            return;
+        }
 
-A functional component that provides a context for a dotCMS page.
+        sendMessageToEditor({ action: DotCMSUVEAction.CLIENT_READY });
+        const subscription = createUVESubscription(UVEEventType.CONTENT_CHANGES,(pageAsset) => setPageAsset(pageAsset));
```

Comment:
should we leave an example with raw strings? That can be useful for JS developers

Reply:
Done! Thanks for that catch! 🙌